File: snippet/examples/ocaml/har.ml
open Cohttp_lwt_unix
open Cohttp
open Lwt

let uri = Uri.of_string "http://localhost:8081/api/pixels" in
let headers = Header.add_list (Header.init ()) [
  ("Content-Type", "application/json");
  ("User-Agent", "curl/7.68.0");
  ("Accept", "*/*");
] in
let body = Cohttp_lwt_body.of_string "{\"x\":1, \"y\":2, \"hex\":\"#102932\"}" in

Client.call ~headers ~body `POST uri
>>= fun (res, body_stream) ->
  (* Do stuff with the result *)

File: snippet/examples/rust/har.rs
use serde_json::json;
use reqwest;

#[tokio::main]
pub async fn main() {
    let url = "http://localhost:8081/api/pixels";

    let payload = json!({
        "x": 1,
        "y": 2,
        "hex": "#102932"
    });

    let mut headers = reqwest::header::HeaderMap::new();
    headers.insert("Content-Type", "application/json".parse().unwrap());
    headers.insert("User-Agent", "curl/7.68.0".parse().unwrap());
    headers.insert("Accept", "*/*".parse().unwrap());

    let client = reqwest::Client::new();
    let response = client.post(url)
        .headers(headers)
        .json(&payload)
        .send()
        .await;

    let results = response.unwrap()
        .json::<serde_json::Value>()
        .await
        .unwrap();

    dbg!(results);
}
